webapp/channels/src/utils/utils.tsx

```diff
@@ -1366,86 +1323,8 @@ export async function handleFormattedTextClick(e: React.MouseEvent, currentRelat
-                            }
-                        } else { // Handle permalink - Get channel data from post
-                            const {postId} = match;
-                            let post = getPost(state, postId!);
-                            if (!post) {
-                                const {data: postData} = await store.dispatch(getPostAction(match.postId!));
-                                post = postData;
-                            }
-                            if (post) {
-                                isReply = Boolean(post.root_id);
-
-                                channel = getChannel(state, post.channel_id);
-                                if (!channel) {
-                                    const {data: channelData} = await store.dispatch(getChannelAction(post.channel_id));
-                                    channel = channelData;
-                                }
-                            }
-                        }
-                        if (channel && channel.type === Constants.PRIVATE_CHANNEL) {
-                            let member = getMyChannelMemberships(state)[channel.id];
-                            if (!member) {
-                                const membership = await store.dispatch(getChannelMember(channel.id, getCurrentUserId(state)));
-                                if ('data' in membership) {
-                                    member = membership.data;
-                                }
-                            }
-                            if (!member) {
-                                const {data} = await store.dispatch(joinPrivateChannelPrompt(team, channel, false));
-                                if (data.join) {
-                                    let error = false;
-                                    if (!getTeamMemberships(state)[team.id]) {
-                                        const joinTeamResult = await store.dispatch(addUserToTeam(team.id, user.id));
-                                        error = joinTeamResult.error;
-                                    }
-                                    if (!error) {
-                                        await store.dispatch(joinChannel(user.id, team.id, channel.id, channel.name));
-                                    }
-                                } else {
-                                    return;
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-
             e.stopPropagation();
-
-            if (match && match.type === 'permalink' && isTeamSameWithCurrentTeam(state, match.teamName) && isReply && crtEnabled) {
-                focusPost(match.postId ?? '', linkAttribute.value, user.id, {skipRedirectReplyPermalink: true})(store.dispatch, store.getState);
-            } else {
-                getHistory().push(linkAttribute.value);
-            }
+            getHistory().push(linkAttribute.value);
```

Comment:
I assume this lets the routing code handle joining a channel now? I don't know why that logic was duplicated here since I don't remember that logic being here before

Reply:
I'm also confused about this. Why did we need all that removed code? And why don't we need it anymore?

Reply:
Not sure why we had it, but ended up calling more or less the same logic twice. Removing it does not seem to create any issue.

Reply:
If you remove this code like you have then clicking cancel on the private channel prompt redirects you to Town Square. With this code the prompt is triggered before any navigation, which allows the user to stay in whichever channel they were in.